extended-it/src/test/java/apoc/vectordb/WeaviateEnterpriseTest.java

```diff
@@ -0,0 +1,151 @@
+package apoc.vectordb;
+
+import apoc.util.MapUtil;
+import apoc.util.Neo4jContainerExtension;
+import apoc.util.TestContainerUtil;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.neo4j.driver.Session;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.weaviate.WeaviateContainer;
+
+import java.util.List;
+import java.util.Map;
+
+import static apoc.ml.RestAPIConfig.HEADERS_KEY;
+import static apoc.util.TestContainerUtil.createEnterpriseDB;
+import static apoc.util.TestContainerUtil.testCall;
+import static apoc.util.TestContainerUtil.testCallEmpty;
+import static apoc.util.TestContainerUtil.testResult;
+import static apoc.util.Util.map;
+import static apoc.vectordb.VectorDbTestUtil.EntityType.FALSE;
+import static apoc.vectordb.VectorDbTestUtil.assertBerlinResult;
+import static apoc.vectordb.VectorDbTestUtil.assertLondonResult;
+import static apoc.vectordb.VectorDbTestUtil.getAuthHeader;
+import static apoc.vectordb.VectorEmbeddingConfig.ALL_RESULTS_KEY;
+import static apoc.vectordb.VectorEmbeddingConfig.FIELDS_KEY;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+
+
+public class WeaviateEnterpriseTest {
+    private static final List<String> FIELDS = List.of("city", "foo");
+    private static final String ADMIN_KEY = "jane-secret-key";
+    private static final String READONLY_KEY = "ian-secret-key";
+    private static final String COLLECTION_NAME = "TestCollection";
+
+    private static final WeaviateContainer WEAVIATE_CONTAINER = new WeaviateContainer("semitechnologies/weaviate:1.24.5")
+            .withEnv("AUTHENTICATION_APIKEY_ENABLED", "true")
+            .withEnv("AUTHENTICATION_APIKEY_ALLOWED_KEYS", ADMIN_KEY + "," + READONLY_KEY)
+            .withEnv("AUTHENTICATION_APIKEY_USERS", "jane@doe.com,ian-smith")
+
+            .withEnv("AUTHORIZATION_ADMINLIST_ENABLED", "true")
+            .withEnv("AUTHORIZATION_ADMINLIST_USERS", "jane@doe.com,john@doe.com")
+            .withEnv("AUTHORIZATION_ADMINLIST_READONLY_USERS", "ian-smith,roberta@doe.com");
+
+    private static final Map<String, String> ADMIN_AUTHORIZATION = getAuthHeader(ADMIN_KEY);
+    private static final Map<String, String> READONLY_AUTHORIZATION = getAuthHeader(READONLY_KEY);
+    private static final Map<String, Object> ADMIN_HEADER_CONF = map(HEADERS_KEY, ADMIN_AUTHORIZATION);
+
+    private static final String ID_1 = "8ef2b3a7-1e56-4ddd-b8c3-2ca8901ce308";
+    private static final String ID_2 = "9ef2b3a7-1e56-4ddd-b8c3-2ca8901ce308";
```

Comment:
Farei un WeaviateTestUtil in cui metterei tutte ste cose qua per riutilizzarle in questo test ed in WeaviateTest

Reply:
Posso metterlo in extended dato che entrambi i test sono in extended oppure devo metterlo in extended-it?

Reply:
@gmarcostam sì sì va bene in extended-it